android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapView.java

```diff
@@ -249,11 +250,23 @@ public AbstractMapFeature getFeatureAt(int i) {
         return mFeatures.get(i);
     }
 
-    public void dispose() {
+    public synchronized void dispose() {
+        ReactContext reactContext = (ReactContext) mContext;
+        reactContext.removeLifecycleEventListener(mLifeCycleListener);
+
         if(mLocationLayer != null){
             mLocationLayer.onStop();
         }
+
         mLocationManger.dispose();
+
+        final RCTMGLMapView mapView = this;
+        reactContext.runOnUiQueueThread(new Runnable() {
+            @Override
+            public void run() {
+                mapView.onDestroy();
+            }
+        });
```

Comment:
Because 'this' is a view, you could just post on its message queue. You can also just call the onDestroy method directly. 

```
post(new Runnable() {
            @Override
            public void run() {
                onDestroy();
            }
        });
```

Reply:
o yeah duh, I was thinking I was in the manager here I'll call it directly 👍